test/dot_index_test.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import numpy
+import random
+from common import TestCase
+from annoy import AnnoyIndex
+
+
+def dot_metric(a, b):
+    return -numpy.dot(a, b)
+
+
+def similarity(a, b):
+    # Could replace this with kendall-tau if we're comfortable
+    # bringing in scipy as a test dependency.
+    return float(len(set(a) & set(b))) / float(len(set(a) | set(b)))
```

Comment:
this looks like jaccard similarity to me? anyway why is that needed? in all the other tests we use recall